src/core/MOM.F90

```diff
@@ -1920,6 +1918,10 @@ subroutine initialize_MOM(Time, param_file, dirs, CS, Time_in)
   endif
   call callTree_waypoint("state variables allocated (initialize_MOM)")
 
+  ! Initialise the diag mediator, this needs to occur _after_ CS%h has been
+  ! allocated.
+  call diag_mediator_init(G, CS%h, param_file, diag)
```

Comment:
How about providing a method to associate h, instead of moving this call so far?